isign/signer.py

```diff
@@ -158,29 +158,31 @@ class HashEntry(asn1crypto.core.Sequence):
                 _fields = [("ident", asn1crypto.core.ObjectIdentifier),
                            ("value", asn1crypto.core.OctetString)]
 
-            for i, entry in enumerate(signer_info['signed_attrs'][3][1]):
-                parsed = entry.parse()
-
-                if parsed[0].native == SHA1_OID:
-                    for cd_hash in cd_hashes:
-                        if cd_hash['hashType'] == SHA1_HASHTYPE:
-                            val = cd_hash[SHA1_HASHTYPE]
-                elif parsed[0].native == SHA256_OID:
-                    for cd_hash in cd_hashes:
-                        if cd_hash['hashType'] == SHA256_HASHTYPE:
-                            val = cd_hash[SHA256_HASHTYPE]
-                else:
-                    raise ValueError('unexpected entry %s' % parsed[0].native)
-                signer_info['signed_attrs'][3][1][i] = asn1crypto.core.Any(
-                    HashEntry({"ident": parsed[0],
-                               "value": asn1crypto.core.OctetString(val)}))
+            if len(signer_info['signed_attrs']) > 3:
```

Comment:
so the idea here is that this is a telltale sign of the old-style stuff

Reply:
Not just a telltale sign, but also we're about to rewrite a new field that, for old-style stuff, doesn't exist. So, this says "only rewrite it if it exists".

A good comment might be something like "Older signatures lack some fields. For now, we don't attempt to add them if they're not already there."